lib/krikri/harvesters/api_harvester.rb

```diff
@@ -0,0 +1,164 @@
+module Krikri::Harvesters
+  ##
+  # A harvester implementation for REST APIs. The default ApiHarvester expects
+  # Solr-like JSON responses/records.
+  #
+  # An internal interface is provided for easier subclassing. A new API
+  # harvester may reimplement:
+  #   - #get_docs (to retrieve record docs from a response)
+  #   - #get_count (to determine total record count from a response)
+  #   - #get_identifier (to retrieve an indentifier from a record document)
+  #   - #get_content (to retrieve a content string from a record document)
+  #   - #next_options` (to generate the parameters for the next request)
+  #
+  # If the content type of the records is other than JSON, you will also want
+  # to override `#content_type`.
+  class ApiHarvester
+    include Krikri::Harvester
+
+    attr_reader :opts
+
+    ##
+    # @param opts [Hash] options for the harvester
+    # @see .expected_opts
+    def initialize(opts = {})
+      super
+      @opts = opts.fetch(:api, {})
+    end
+
+    ##
+    # @return [Hash] A hash documenting the allowable options to pass to
+    #   initializers.
+    #
+    # @see Krikri::Harvester::expected_opts
+    def self.expected_opts
+      {
+        key: :api,
+        opts: {
+          params: { type: :string, required: false }
```

Comment:
It looks like `:params` might want to be specified as a Hash, because of the way it's dereferenced in `#next_options`